lib/Parser/ptree.cpp

```diff
@@ -0,0 +1,258 @@
+}
+
+ParseNodeBlock * ParseNode::AsParseNodeBlock()
+{
+    Assert(this->nop == knopBlock);
+    return (ParseNodeBlock*)this;
+}
+
+ParseNodeJump * ParseNode::AsParseNodeJump()
+{
+    Assert(this->nop == knopBreak || this->nop == knopContinue);
+    return (ParseNodeJump*)this;
+}
+
+ParseNodeWith * ParseNode::AsParseNodeWith()
+{
+    Assert(this->nop == knopWith);
+    return (ParseNodeWith*)this;
+}
+
+ParseNodeIf * ParseNode::AsParseNodeIf()
+{
+    Assert(this->nop == knopIf);
+    return (ParseNodeIf*)this;
+}
+
+ParseNodeSwitch * ParseNode::AsParseNodeSwitch()
+{
+    Assert(this->nop == knopSwitch);
+    return (ParseNodeSwitch*)this;
+}
+
+ParseNodeCase * ParseNode::AsParseNodeCase()
+{
+    Assert(this->nop == knopCase);
+    return (ParseNodeCase*)this;
+}
+
+ParseNodeReturn * ParseNode::AsParseNodeReturn()
+{
+    Assert(this->nop == knopReturn);
+    return (ParseNodeReturn*)this;
+}
+
+ParseNodeTryFinally * ParseNode::AsParseNodeTryFinally()
+{
+    Assert(this->nop == knopTryFinally);
+    return (ParseNodeTryFinally*)this;
+}
+
+ParseNodeTryCatch * ParseNode::AsParseNodeTryCatch()
+{
+    Assert(this->nop == knopTryCatch);
+    return (ParseNodeTryCatch*)this;
+}
+
+ParseNodeTry * ParseNode::AsParseNodeTry()
+{
+    Assert(this->nop == knopTry);
+    return (ParseNodeTry*)this;
+}
+
+ParseNodeCatch * ParseNode::AsParseNodeCatch()
+{
+    Assert(this->nop == knopCatch);
+    return (ParseNodeCatch*)this;
+}
+
+ParseNodeFinally * ParseNode::AsParseNodeFinally()
+{
+    Assert(this->nop == knopFinally);
+    return (ParseNodeFinally*)this;
+}
+
+ParseNodeLoop * ParseNode::AsParseNodeLoop()
+{
+    Assert(this->nop == knopWhile || this->nop == knopDoWhile || this->nop == knopFor || this->nop == knopForIn || this->nop == knopForOf);
+    return (ParseNodeLoop*)this;
+}
+
+ParseNodeWhile * ParseNode::AsParseNodeWhile()
+{
+    Assert(this->nop == knopWhile || this->nop == knopDoWhile);
+    return (ParseNodeWhile*)this;
+}
+
+ParseNodeFor * ParseNode::AsParseNodeFor()
+{
+    Assert(this->nop == knopFor);
+    return (ParseNodeFor*)this;
+}
+
+ParseNodeForInOrForOf * ParseNode::AsParseNodeForInOrForOf()
+{
+    Assert(this->nop == knopForIn || this->nop == knopForOf);
+    return (ParseNodeForInOrForOf*)this;
+}
+
+ParseNodeFnc * ParseNode::AsParseNodeFnc()
+{
+    Assert(this->nop == knopFncDecl || this->nop == knopProg || this->nop == knopModule);
+    return (ParseNodeFnc*)this;
+}
+
+ParseNodeProg * ParseNode::AsParseNodeProg()
+{
+    Assert(this->nop == knopProg);
+    return (ParseNodeProg*)this;
+}
+
+ParseNodeModule * ParseNode::AsParseNodeModule()
+{
+    // TODO: Currently there is not way to distingish a ParseNodeModule to ParseNodeProg node
+    Assert(this->nop == knopProg);
+    return (ParseNodeModule*)this;
+}
+
+IdentPtr ParseNode::name()
+{
+    if (this->nop == knopName || this->nop == knopStr)
+    {
+        return this->AsParseNodePid()->pid;
+    }
+    else if (this->nop == knopVarDecl)
```

Comment:
Can this be `else if (this->nop == knopVarDecl || this->nop == knopConstDecl)` ? Looks like the two cases are the same.

Reply:
Yeah.  This was just copied over from ptree.h. So it was like that before.  I have cleaned it up.

---
In reply to: [171452685](https://github.com/Microsoft/ChakraCore/pull/4763#discussion_r171452685) [](ancestors = 171452685)